aptos-move/aptos-vm/src/move_vm_ext/write_op_converter.rs

```diff
@@ -39,24 +42,61 @@ impl<'r> WriteOpConverter<'r> {
         }
     }
 
-    pub(crate) fn convert(
+    pub(crate) fn convert_resource(
         &self,
         state_key: &StateKey,
         move_storage_op: MoveStorageOp<Bytes>,
         legacy_creation_as_modification: bool,
+    ) -> Result<WriteOp, VMStatus> {
+        self.convert(
+            self.remote.get_resource_state_value_metadata(state_key),
+            move_storage_op,
+            legacy_creation_as_modification,
+        )
+    }
+
+    pub(crate) fn convert_module(
+        &self,
+        state_key: &StateKey,
+        move_storage_op: MoveStorageOp<Bytes>,
+        legacy_creation_as_modification: bool,
+    ) -> Result<WriteOp, VMStatus> {
+        self.convert(
+            self.remote.get_module_state_value_metadata(state_key),
+            move_storage_op,
+            legacy_creation_as_modification,
+        )
+    }
+
+    pub(crate) fn convert_aggregator(
+        &self,
+        state_key: &StateKey,
+        move_storage_op: MoveStorageOp<Bytes>,
+        legacy_creation_as_modification: bool,
+    ) -> Result<WriteOp, VMStatus> {
+        self.convert(
+            self.remote
+                .get_aggregator_v1_state_value_metadata(state_key),
+            move_storage_op,
+            legacy_creation_as_modification,
+        )
+    }
+
+    pub fn convert(
```

Comment:
why is convert pub now? used to be pub(crate).

Also, another way to do this would be to have an enum 
ContentKind {
Resource
Module
Aggregator
}, pass that to convert, match and call one of the 3 methods, but whatever feels cleaner to you - they may not be enough duplication to fully warrant this.

Reply:
Messed up the merge... and it became pub. I also wanted to use enums before, but then we dispatch in the things we already know, which seemed a bit annoying given that this is called for every session. Added macros to reduce duplication :)